plugins/c8y_configuration_plugin/src/main.rs

```diff
@@ -134,73 +135,71 @@ async fn run(
     );
     let () = mqtt_client.published.send(msg).await?;
 
-    // Mqtt message loop
-    while let Some(message) = mqtt_client.received.next().await {
-        debug!("Received {:?}", message);
-        if let Ok(payload) = message.payload_str() {
-            let result = if let "tedge/configuration_change/c8y-configuration-plugin" =
-                message.topic.name.as_str()
-            {
-                // Reload the plugin config file
-                plugin_config = PluginConfig::new(config_file_path);
-                // Resend the supported config types
-                let msg = plugin_config.to_supported_config_types_message()?;
-                mqtt_client.published.send(msg).await?;
-                Ok(())
-            } else {
-                match payload.split(',').next().unwrap_or_default() {
-                    "524" => {
-                        let maybe_config_download_request =
-                            SmartRestConfigDownloadRequest::from_smartrest(payload);
-                        if let Ok(config_download_request) = maybe_config_download_request {
-                            handle_config_download_request(
-                                &plugin_config,
-                                config_download_request,
-                                tmp_dir.clone(),
-                                &mut mqtt_client,
-                                http_client,
-                            )
-                            .await
-                        } else {
-                            error!("Incorrect Download SmartREST payload: {}", payload);
-                            Ok(())
-                        }
-                    }
-                    "526" => {
-                        // retrieve config file upload smartrest request from payload
-                        let maybe_config_upload_request =
-                            SmartRestConfigUploadRequest::from_smartrest(payload);
-
-                        if let Ok(config_upload_request) = maybe_config_upload_request {
-                            // handle the config file upload request
-                            handle_config_upload_request(
-                                &plugin_config,
-                                config_upload_request,
-                                &mut mqtt_client,
-                                http_client,
-                            )
-                            .await
-                        } else {
-                            error!("Incorrect Upload SmartREST payload: {}", payload);
-                            Ok(())
-                        }
-                    }
-                    _ => {
-                        // Ignore operation messages not meant for this plugin
-                        Ok(())
+    let fs_notification_stream = fs_notify_stream(&[(
+        config_dir,
+        Some(config_file.to_string()),
+        &[FileEvent::Modified, FileEvent::Deleted, FileEvent::Created],
+    )])?;
```

Comment:
> * Is `config_file` equals to `c8y/c8y-configuration-plugin.toml` or `c8y-configuration-plugin.toml` ?
> * In other words, does the `fs_notify_stream` watch the directory recursively i.e including the sub-directories as `c8y`?

No it only watches for `c8y-configuration-plugin.toml` inside the c8y dir. But it supports a relative path to the file. Which is handled [here](https://github.com/thin-edge/thin-edge.io/blob/325935ef3c7c17d3c43e719211a410407a734cb3/crates/common/tedge_utils/src/fs_notify.rs#L141-L151).

Reply:
> Another comment about removing the `--config-file` option. Are you proposing to remove this option because of this call where you need a directory and some file to watch?

No only to keep the consistency across both log and configuration plugins. By design that helper function takes those arguments as inputs but it could have been abstracted in the API to take a full path to a file for a full path to a directory.